R/permutation.R

```diff
@@ -0,0 +1,473 @@
+        ggplot2::labs(color = NULL, y = 'Estimate') +
+        ggplot2::guides(linetype = 'none') +
+        ggplot2::theme_bw() +
+        ggplot2::theme(legend.position = 'bottom')
+
+    return(out_plot)
+}
+
+
+#' Generate augsynth plot with 95\% CI based on inference preference
+#'
+#' @param results Results from calling xxx
+#' @param inf_type Inference type (takes a value of 'permutation' or 'permutation_rstat')
+#'
+ci95_rstat_plot <- function(augsynth, inf_type) {
+
+    stopifnot(inf_type %in% c('permutation', 'permutation_rstat'))
+
+    if(inf_type == 'permutation') {
+        measure = "ATT"
+    } else {
+        measure = 'rstat'
+    }
+
+    if (is.null(augsynth$results$permutations)) {
+        augsynth <- add_placebo_distribution(augsynth)
+    }
+
+    plot_df <- augsynth$results$permutations$MDES_table
+
+    if (measure == 'ATT') {
+        plot_df <- plot_df %>%
+            mutate(lower = ATT + (qnorm(0.025) * SE_gap),
+                   upper = ATT + (qnorm(0.975) * SE_gap),
+            )
+    } else if (measure == 'rstat') {
+        plot_df <- plot_df %>%
+            mutate(lower = ATT + (qnorm(0.025) * SE_rstat),
+                   upper = ATT + (qnorm(0.975) * SE_rstat),
+            )
+    }
+
+    t0 <- ncol(augsynth$data$X)
+    treat_year = augsynth$data$time[t0 + 1]
+
+    outplot <- ggplot2::ggplot(data = plot_df, aes(x = !!as.name(augsynth$time_var))) +
+        ggplot2::geom_ribbon(data = plot_df %>% filter(!!as.name(augsynth$time_var) >= treat_year),
+                             aes(ymin = lower, ymax = upper), alpha = 0.2) +
+        ggplot2::geom_line(aes(y = ATT)) +
+        ggplot2::geom_vline(lty = 2, xintercept = treat_year) +
+        ggplot2::geom_hline(lty = 2, yintercept = 0) +
+        ggplot2::scale_color_manual(values = c('Control' = 'gray', 'Treatment' = 'black')) +
+        ggplot2::labs(color = NULL, y = 'Estimate') +
+        ggplot2::theme_bw()
+
+    return(outplot)
+}
+
+
+
+#' Generate formatted outputs for statistical inference using permutation inference
+#'
+#' @param augsynth An augsynth object.
+#'
+#' @noRd
+permutation_inf <- function(augsynth, inf_type) {
+
+    t0 <- dim(augsynth$data$synth_data$Z0)[1]
+    tpost <- dim(augsynth$data$synth_data$Z0)[1]
+
+    out <- list()
+    out$att <- augsynth$results$permutations$MDES_table$ATT
+
+    SEg = NA
+    if (inf_type == 'permutation') {
+        SEg = augsynth$results$permutations$MDES_table$SE_gap
+        pval = augsynth$results$permutations$MDES_table$p_gap
+    } else if (inf_type == 'permutation_rstat') {
+        SEg = augsynth$results$permutations$MDES_table$SE_rstat
+        pval = augsynth$results$permutations$MDES_table$p_rstat
+    }
+    out$lb <- out$att + (qnorm(0.025) * SEg)
+    out$ub <- out$att + (qnorm(0.975) * SEg)
+    out$p_val <- pval
+
+    out$lb[c(1:t0)] <- NA
+    out$ub[c(1:t0)] <- NA
+    out$p_val[c(1:t0)] <- NA
+    out$alpha <- 0.05
+
+    return(out)
```

Comment:
The code for getting permutation inference has changed since this example, but if I remember correctly NAs were being returned for pre-treatment p-values (but not for all outcomes). This is still the case if you run summary(syn, inf_type = 'permutation'). If you scroll down to the post-treatment time periods, should be non-NA p-values.